### PR TITLE
Use Almalinux as base image, and upgrade vespa to the latest version

### PR DESCRIPTION
**Pull request by papa99do**
* **What is the current behavior?**  
* the base image centos:stream8 is end of life cycle. 
* we are on Vespa 8.332.5
* use different way to install ffmpeg based on platform

* **What is the new behavior?**
* use almalinux8 as the new base image
* Upgrade vespa to 8.396.18
* use consolidated way to install ffmpeg 

* **Have tests been run against this PR?**  
* https://github.com/marqo-ai/marqo/actions/runs/10517403358

* **Have appropriate comments and documentation been made on this PR?**  

* **Related changes in other repos** (link commit/PR here)

* **Other information**:

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,44 +1,29 @@
...
+# Update the public key
+RUN rpm --import https://repo.almalinux.org/almalinux/RPM-GPG-KEY-AlmaLinux-8
 
 # Install base packages that are used across both the application and Vespa
 RUN dnf install -y epel-release dnf-utils ca-certificates curl gnupg && \
     dnf config-manager --set-enabled powertools
 
-
 # Install application specific packages
-RUN dnf groupinstall "Development Tools" -y && \
```
Comment (papa99do) [inline Dockerfile:12]: only gcc is used in installation of python packages. This also introduce java 1.8 dep which messes with the openjdk-17 installation.

File: Dockerfile [hunk 2]
```diff
@@ -1,44 +1,29 @@
...
+        gcc \
+        jq \
+        unzip \
+        tmux
 
 # Set up Python 3.8 and pip
 RUN alternatives --set python3 /usr/bin/python3.8 && \
     curl https://bootstrap.pypa.io/get-pip.py | python3
 
-# Punkt Tokenizer setup
```
Comment (papa99do) [inline Dockerfile:36]: moved to a separate script

File: Dockerfile [hunk 3]
```diff
@@ -47,23 +32,20 @@ RUN pip3 install --no-cache-dir -r requirements.txt
 COPY scripts scripts
 RUN bash scripts/install_onnx_gpu_for_amd.sh && \
     bash scripts/install_torch_amd.sh && \
-    bash scripts/install_redis_centos.sh && \
-    mkdir -p /etc/redis && \
-    touch /etc/redis/redis.conf && \
-    echo "echo never > /sys/kernel/mm/transparent_hugepage/enabled" >> /etc/rc.local && \
-    echo "save ''" >> /etc/redis/redis.conf
```
Comment (papa99do) [inline Dockerfile:54]: moved to install_redis.sh script

File: Dockerfile [hunk 4]
```diff
@@ -47,23 +32,20 @@ RUN pip3 install --no-cache-dir -r requirements.txt
...
-RUN dnf config-manager \
-  --add-repo https://raw.githubusercontent.com/vespa-engine/vespa/master/dist/vespa-engine.repo &&\
-    dnf config-manager --enable powertools &&\
-    dnf install -y epel-release &&\
-    dnf install -y vespa
+ADD scripts/start_vespa.sh /usr/local/bin/start_vespa.sh
 
 # Set Envs for Vespa
 ENV PATH="/opt/vespa/bin:/opt/vespa-deps/bin:${PATH}"
+# TODO check if following env vars are required
```
Comment (papa99do) [inline Dockerfile:48]: @wanliAlex , are these two env vars used? `VESPA_LOG_STDOUT` and `VESPA_LOG_FORMAT`?

File: scripts/install_redis.sh [hunk 5]
```diff
@@ -1,10 +1,11 @@
 #!/bin/bash
 # This script is meant to be run at buildtime.
 
-# These updates must be run, or else old redis could be installed (like version 5.0.7)
```
Comment (papa99do) [inline scripts/install_redis.sh:4]: There is a legacy script to install redis to ubuntu, probably left over from v1. I've deleted that script, and renamed `install_redis_centos.sh` to `install_redis.sh`. Therefore the diff is a bit confusing.

File: .github/workflows/build_push_dev_img.yml [hunk 6]
```diff
@@ -27,7 +27,7 @@ jobs:
           aws-region: us-west-2
       - name: Start EC2 runner
         id: start-ec2-runner
-        uses: machulav/ec2-github-runner@v2
+        uses: machulav/ec2-github-runner@v3
```
Comment (papa99do) [inline .github/workflows/build_push_dev_img.yml:30]: @farshidz , you mentioned that we should ping this to a commit hash. Should we ping it to the commit of v2 version we used previously or the commit of the latest release version?

File: Dockerfile (hunk 4 above)
Comment (papa99do) [reply Dockerfile:49]: I'll leave them here for now. and remove them in the next release of marqo-base image

**Review by wanliAlex — APPROVED** [review]
Looks good to me

**Pull request merged**